GeometricSeqPhenotype.java

```diff
@@ -0,0 +1,308 @@
+     */
+    public Phenotype mutate() {
+        // Implementation:
+        // Mutates a nucleotide in the sequence at a random index
+        // If mutation creates a stop codon, then mutate at another index
+
+        // this.nucleotideSequence is represented using a char[] instead of
+        //  - String since Strings are immutable. New Strings have to be created to mutate a sequence,
+        //    which is messy and uses up extra heap space. Plus, Strings include an extra byte '\n'.
+        //  - StringBuffer because it's just a Wrapper around a char[], and has functionality that we don't need such as resizing.
+
+        String wildTypeAminoAcid = "", mutantAminoAcid = "";
+        int mutationIndexSiteSequence = -1;
+        char mutatedNucleotide = ' ', originalNucleotide = ' ';
+
+        // mutate at least once
+        // keep mutating if mutation generates a stop codon
+        while (mutantAminoAcid.equals("") || mutantAminoAcid.equals("STOP")) {
+            // choose random index to mutate in this.nucleotideSequence
+            mutationIndexSiteSequence = random(this.nucleotideSequence.length);
+
+            originalNucleotide = this.nucleotideSequence[mutationIndexSiteSequence];
+
+            char originalNucleotideToMutate = this.nucleotideSequence[mutationIndexSiteSequence];
+
+            // get mutant nucleotide (transition transversion ratio)
+            mutatedNucleotide = Biology.MutationType.MUTATION.getNucleotide(originalNucleotideToMutate);
+
+            String[] wildTypeMutantAminoAcids = mutateHelper(mutationIndexSiteSequence, mutatedNucleotide);
+
+            wildTypeAminoAcid = wildTypeMutantAminoAcids[0];
+            mutantAminoAcid = wildTypeMutantAminoAcids[1];
+        }
+
+        int mutationIndexSite = mutationIndexSiteSequence / 3; // site # where mutation is occurring {0, . . ., total number of sites - 1}
+
+        // update parameter 1 and 2
+        double[] vectors = getVectors(mutationIndexSite, wildTypeAminoAcid, mutantAminoAcid);
+        double mutA = vectors[0];
+        double mutB = vectors[1];
+
+        // update parameter 3
+        // get new nucleotide sequence after mutating at the random index
+        char[] copyNucleotideSequence = Arrays.copyOf(this.nucleotideSequence, Parameters.startingSequence.length());
+        copyNucleotideSequence[mutationIndexSiteSequence] = mutatedNucleotide;
+
+        // update parameter 4 and 5
+        // count number of epitope and non-epitope mutations
+        int eMutationNew = this.eMutation;
+        int neMutationNew = this.neMutation;
+        if (Biology.SiteMutationVectors.VECTORS.getEpitopeSites().contains(mutationIndexSite)) {
+            eMutationNew += 1;
+        } else {
+            neMutationNew += 1;
+        }
+
+        // uncomment line below to test transition transversion ratio:
+        // TestGeometricSeqPhenotype.mutations.println("" + originalNucleotide + mutatedNucleotide);
```

Comment:
Can we remove these lines?

Reply:
`testTransitionTransversion.py` depends on the CSV file produced by line 229. These 2 lines are for us to test our code. Do you think it's okay if we remove it now?